diff --git a/ajax_poll.module b/ajax_poll.module
index fa073b8..3cfc650 100644
--- a/ajax_poll.module
+++ b/ajax_poll.module
@@ -84,7 +84,7 @@ function ajax_poll_callback($type, $node, $teaser, $block) {
   poll_view($node, 'full');
 
   // Reset POST and the $node so that we get fresh copies.
-  unset($_POST);
+  $_POST = array();
   $node = node_load($node->nid, NULL, TRUE);
 
   $view_mode = $teaser ? 'teaser' : 'full';
