diff --git a/better_exposed_filters_exposed_form_plugin.inc b/better_exposed_filters_exposed_form_plugin.inc
index 3b6afb6..a4bbbf1 100644
--- a/better_exposed_filters_exposed_form_plugin.inc
+++ b/better_exposed_filters_exposed_form_plugin.inc
@@ -1117,9 +1117,8 @@ dateFormat: "dd-mm-yy"
         // Collect replacement data.
         $data = array();
         $available = $options['more_options']['tokens']['available'];
-        if (in_array('vocabulary', $available)) {
-          $vocabs = taxonomy_get_vocabularies();
-          $data['vocabulary'] = $vocabs[$filters[$label]->options['vid']];
+        if (in_array('vocabulary', $available) && isset($filters[$label]->definition['vocabulary'])) {
+          $data['vocabulary'] = taxonomy_vocabulary_machine_name_load($filters[$label]->definition['vocabulary']);
         }
         /* Others? */

